pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/SegmentMetadataImpl.java

```diff
@@ -72,6 +72,7 @@
   private boolean _hasStarTree;
   private StarTreeMetadata _starTreeMetadata = null;
   private String _creatorName;
+  private char _paddingCharacter = V1Constants.Str.DEFAULT_STRING_PAD_CHAR;
```

Comment:
this should be set to LEGACY padding character until the server code is pushed, so that in case a segment is generated it will always be the same as the old ones (except it will have the extra metadata), and the old servers will be able to read them.
Of course, you should test that the old servers can still read the column metadata that has this extra property.